File: no-sdk/dotnet/HttpHelpers.cs
public static class HttpHelpers
{
    public static string Get(string service, string region, string url)
    {
        var uri = new Uri(url);        
        var now = DateTime.UtcNow;
        var amzDate = AWSSigner.ToAmzDate(now);
        var authorizationHeader = AWSSigner.GetAuthorizationHeader(service, region,"GET", uri, now);

        // Make the request
        using var client = new HttpClient();
        client.DefaultRequestHeaders.Add("Host", uri.Host);
        client.DefaultRequestHeaders.Add("x-amz-date", amzDate);
        if(!string.IsNullOrWhiteSpace(AwsCredentials.AWS_SESSION_TOKEN))
        {
            client.DefaultRequestHeaders.Add("x-amz-security-token", AwsCredentials.AWS_SESSION_TOKEN);
        }        
        client.DefaultRequestHeaders.TryAddWithoutValidation("Authorization", authorizationHeader);

        var response = client.GetAsync(uri).Result;
        
        var responseBody = response.IsSuccessStatusCode
            ? response.Content.ReadAsStringAsync().Result
            : $"Error: {(int)response.StatusCode} {response.ReasonPhrase} {response.Content.ReadAsStringAsync().Result}";
        
        return responseBody;
    }

    public static string Post(string service, string region, string url, string payload)
    {
        var uri = new Uri(url);        
        var now = DateTime.UtcNow;
        var amzDate = AWSSigner.ToAmzDate(now);
        var payloadHash = AWSSigner.CalculateHash(payload);
        var headers = new Dictionary<string, string>
        {
            {"x-amz-content-sha256", payloadHash},
            {"content-length", payload.Length.ToString()},
            {"content-type", "application/json"}
        };
        var authorizationHeader = AWSSigner.GetAuthorizationHeader(service, region,"POST", uri, now, headers, payloadHash);

        // Make the request
        using var client = new HttpClient();
        client.DefaultRequestHeaders.Add("Host", uri.Host);
        client.DefaultRequestHeaders.Add("x-amz-date", amzDate);
        if(!string.IsNullOrWhiteSpace(AwsCredentials.AWS_SESSION_TOKEN))
        {
            client.DefaultRequestHeaders.Add("x-amz-security-token", AwsCredentials.AWS_SESSION_TOKEN);
        }        
        client.DefaultRequestHeaders.TryAddWithoutValidation("Authorization", authorizationHeader);

        var requestContent = new StringContent(payload);
        requestContent.Headers.Remove("content-type");
        requestContent.Headers.Add("content-type", "application/json");
        requestContent.Headers.Add("content-length", payload.Length.ToString());
        requestContent.Headers.Add("x-amz-content-sha256", payloadHash);

        var response = client.PostAsync(uri, requestContent).Result;
        
        var responseBody = response.IsSuccessStatusCode
            ? response.Content.ReadAsStringAsync().Result
            : $"Error: {(int)response.StatusCode} {response.ReasonPhrase} {response.Content.ReadAsStringAsync().Result}";
        
        return responseBody;
    }
}

File: no-sdk/java/HttpHelpers.java
package sigv4.signing;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URI;
import java.net.URISyntaxException;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.ZonedDateTime;
import java.util.HashMap;
import java.util.Map;

public class HttpHelpers {

    public static String get(String service, String region, String requestUrl) throws IOException, InterruptedException, URISyntaxException {
        URI uri = new URI(requestUrl);
        ZonedDateTime now = ZonedDateTime.now();
        String amzDate = AWSSigner.toAmzDate(now);

        String authorizationHeader = AWSSigner.getAuthorizationHeader(service, region, "GET", uri, now);

        HttpClient httpClient = HttpClient.newHttpClient();
        HttpRequest.Builder requestBuilder = HttpRequest.newBuilder()
                .uri(URI.create(requestUrl))
                .header("x-amz-date", amzDate)
                .header("Authorization", authorizationHeader);
        if (AwsCredentials.getAwsSessionToken() != null && !AwsCredentials.getAwsSessionToken().isEmpty()) {
            requestBuilder.header("x-amz-security-token", AwsCredentials.getAwsSessionToken());
        }
        HttpRequest request = requestBuilder.build();

        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        if (response.statusCode() == HttpURLConnection.HTTP_OK){
            return response.body().toString();
        }else{
            return String.format("Error: %d %s", response.statusCode(), response.body().toString());
        }    
    }

    public static String post(String service, String region, String requestUrl, String payload) throws IOException, InterruptedException, URISyntaxException {
        URI uri = new URI(requestUrl);
        ZonedDateTime now = ZonedDateTime.now();
        String amzDate = AWSSigner.toAmzDate(now);

        String payloadHash = AWSSigner.calculateHash(payload);
        Map<String, String> headers = new HashMap<>();
        headers.put("x-amz-content-sha256", payloadHash);
        headers.put("content-length", String.valueOf(payload.length()));
        headers.put("content-type", "application/json");

        String authorizationHeader = AWSSigner.getAuthorizationHeader(service, region, "POST", uri, now, headers, payloadHash);

        HttpClient httpClient = HttpClient.newHttpClient();
        HttpRequest.Builder requestBuilder = HttpRequest.newBuilder()
                .uri(URI.create(requestUrl))
                .header("x-amz-date", amzDate)
                .header("Authorization", authorizationHeader)
                .header("x-amz-content-sha256", payloadHash)
                .header("content-type", "application/json");

        if (AwsCredentials.getAwsSessionToken() != null && !AwsCredentials.getAwsSessionToken().isEmpty()) {
            requestBuilder.header("x-amz-security-token", AwsCredentials.getAwsSessionToken());
        }

        HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(payload)).build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        if (response.statusCode() == HttpURLConnection.HTTP_OK){
            return response.body().toString();
        }else{
            return String.format("Error: %d %s", response.statusCode(), response.body().toString());
        }
    }
}